### ISSUE TITLE
reportShadowedImports is warning about a relative package import

### ISSUE THREAD

**Issue by joerick** [microsoft/pyright#4269]
**Describe the bug**
reportShadowedImports is warning about a relative package import. For example- 

**To Reproduce**
Steps to reproduce the behavior.

**Expected behavior**

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
def foo(self) -> str:
    return 3
```

If your code relies on symbols that are imported from a third-party library, include the associated import statements and specify which versions of those libraries you have installed.

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool? Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.

**Additional context**

**Comment by rchiodo** [comment]
This should be relatively easy to fix for the import line (check if it's relative), but it also shows up in the typing.py file too:

![image](https://user-images.githubusercontent.com/19672699/205369303-0f3c0c2c-865d-4ea6-8a7d-5ac3d808a809.png)

The only way to turn that warning off (at the moment) is to disable all errors in that file or add this setting:

```jsonc
"python.analysis.diagnosticSeverityOverrides": {
    "reportShadowedImports": "none"
  },
```

**Comment by joerick** [comment]
Thanks for looking at this. FWIW, I don't see the error in the typing.py file on my project-

<img width="1595" alt="image" src="https://user-images.githubusercontent.com/1244307/205494686-e7266386-f060-4a99-8a6a-6dab0d74134f.png">

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by joerick** [comment]
Thanks!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.283, which I just published. It will also be included in a future version of pylance.
